modules/subnet/subnet.tf

```diff
@@ -0,0 +1,19 @@
+resource "oci_core_subnet" "vcn_subnet" {
+  for_each       = length(var.subnets) > 0 ? var.subnets : {}
+  cidr_block     = each.value.cidr_block
+  compartment_id = var.compartment_id
+  vcn_id         = var.vcn_id
+
+
+  defined_tags = var.defined_tags
+  #dhcp_options_id            = oci_core_dhcp_options.test_dhcp_options.id
+  display_name  = lookup(each.value, "name", each.key)
+  dns_label     = lookup(each.value, "dns_label", null)
+  freeform_tags = var.freeform_tags
+  #ipv6cidr_block             = var.enable_ipv6 == false ? null : each.value.ipv6cidr_block
```

Comment:
add a comment to explain why those lines are commented

Reply:
done .